### ISSUE TITLE
Sending email returns 451 4.7.500 Access denied

### ISSUE THREAD

**Issue by bogdanst24** [jstedfast/MailKit#1502]
We have been running MailKit on a production product for 4+ years already in an SMTP relay connected to Exchange Online.
Starting a few months ago, we noticed a higher and higher increase in mail-sending errors. 
Initially, the errors were mainly coming from emails sent to large (100+ recipients) distribution lists. Now we can see the errors even in emails sent to 1 or 2 recipients. 

In some cases, where we could find more information (as the solution is used by customers), the emails were sent regardless of the error. I know it is a long shot, but alongside some changes in Exchange Online we can't find anything about it, another possibility of the issue could be related to the way MailKit catches and interprets the errors coming from Exchange.

We are not yet able to reproduce the error locally or put our finger on a specific issue. 

_The latest version of MailKit we are using is 3.3.0._ 

The error we are seeing a lot recently:

>  Returned SMTP StatusCode: 451 ErrorInProcessing. System.Exception: 4.7.500 Access denied, please try again later. For more information please go to http://go.microsoft.com/fwlink/?LinkId=526653. (AS112201)
  at void MailKit.Net.Smtp.SmtpClient.OnRecipientNotAccepted(MimeMessage message, MailboxAddress mailbox, SmtpResponse response)
  at bool MailKit.Net.Smtp.SmtpClient.ProcessRcptToResponse(MimeMessage message, MailboxAddress mailbox, SmtpResponse response)
  at async Task MailKit.Net.Smtp.SmtpClient.FlushCommandQueueAsync(MimeMessage message, MailboxAddress sender, IList<MailboxAddress> recipients, bool doAsync, CancellationToken cancellationToken)
  at async Task MailKit.Net.Smtp.SmtpClient.SendAsync(FormatOptions options, MimeMessage message, MailboxAddress sender, IList<MailboxAddress> recipients, bool doAsync, CancellationToken cancellationToken, ITransferProgress progress)
  at async Task MailKit.Net.Smtp.SmtpClient.SendAsync(FormatOptions options, MimeMessage message, MailboxAddress sender, IList<MailboxAddress> recipients, bool doAsync, CancellationToken cancellationToken, ITransferProgress progress)
  at async Task

**Comment by jstedfast** [comment]
According to the fwlink in the exception message:

Error code | Description | Possible cause | Additional information
-- | -- | -- | --
4.7.500-699 | Access denied, please try again later | Suspicious activity has been detected and sending has been temporarily restricted for further evaluation. | If this activity is valid, this restriction will be lifted shortly.

I'm not sure what would constitute as "suspicious activity", but likely some sort of new anti-spam measure was recently put into place on the Exchange server and that is what is causing these failures.

**Comment by jstedfast** [comment]
Neither of those scenarios are possible.

**Comment by bogdanst24** [comment]
After some time I think I have managed to find the fix for it, but I don't understand exactly why.

_Failing scenario I was able to reproduce:_
Send one email to 500 recipients. Because we communicate with Exchange and there are hard limits of recipients in one message, we split it into multiple transactions of 100 recipients each, sent one after another.
It will always be the last transaction (batch of 100) to fail with the exception from above.

_Potential fix I'm trying to understand_
In the original implementation, we were creating the SMTP connection before the first transaction and never disconnected it for the email in the discussion. So all 5 transactions will use the same SmptSession. The `disconnectAsync` was never used, and it would always fail after ~300+ recipients were sent.
What fixes the issue seems to be doing the disconnect after each mail sending, and so always recreating the connection.

I am aware that there might be unknown limits on Exchange Online about the number of recipients/emails per SMTP connection, or something similar. However, I wanted to also verify that the issue was not caused by some miss-implementation of MailKit from our side or a bug in the library.
